main.go

```diff
@@ -102,16 +103,82 @@ func fatal(i interface{}) {
 	os.Exit(1)
 }
 
-func main() {
-	args := os.Args
-	if len(args) < 2 {
-		fmt.Println("Usage: ", args[0], "<testfile>")
-		os.Exit(1)
+// Opts holds the result of parsed input args
+type Opts struct {
+	params Params
+}
+
+// all input options + corresponding data structures should be defined here
+func newOpts() *Opts {
+	opts := &Opts{
+		params: make(Params),
 	}
 
-	var params = make(map[string]string)
-	if len(args) > 2 {
-		params = parseArgs(args[2:])
+	flag.Var(&opts.params, "param",
+		`Replace all test parameter instances of <name> with <value>.
+        Separate multiple `+"`<name>:<value>` inputs with ',' or pass this flag multiple times.")
+
+	return opts
+}
+
+/* parse all input args */
+func parseArgs() (*Opts, []string) {
+	// create opts
+	opts := newOpts()
+	// set usage
+	flag.Usage = usage
+	// parse all args
+	flag.Parse()
+
+	// get all non-flag args
+	rest := append([]string{os.Args[0]}, flag.Args()...)
+
+	return opts, rest
+}
+
+// Params holds the parameter mappings specified  by user
+type Params map[PName]PValue
+type PName string
+type PValue string
+
+// specific Params -> String transformation
+func (params *Params) String() string {
+	return fmt.Sprint(*params)
+}
+
+// tell `flag` package how to parse/store param args
+func (params *Params) Set(value string) error {
+	for _, p := range strings.Split(value, ",") {
+		pElems := strings.SplitN(p, ":", 2)
```

Comment:
@hsanjuan Right, the reason I did `:` is because with `=` you might have something that looks like: `--param=N=3`, which I thought was confusing against `--param=N:3`. The `=` after `--param` isn't required, but is a potential option through the `flag` package. I figured I'd just take away the option and 'make confusing practices difficult/impossible'. Thoughts?